hazelcast/src/main/java/com/hazelcast/scheduledexecutor/impl/DistributedScheduledExecutorService.java

```diff
@@ -183,4 +219,62 @@ private void resetPartitionOrMemberBinContainer(String name) {
             partition.destroyContainer(name);
         }
     }
+
+    private void registerPartitionListener() {
+        this.partitionLostRegistration = getNodeEngine().getPartitionService().addPartitionLostListener(
+                new PartitionLostListener() {
+                    @Override
+                    public void partitionLost(PartitionLostEvent event) {
+                        for (ScheduledFutureProxy future : lossListeners) {
+                            future.notifyPartitionLost(event);
+                        }
+                    }
+                }
+        );
+    }
+
+    private void unRegisterPartitionListenerIfExists() {
+        if (this.partitionLostRegistration == null) {
+            return;
+        }
+
+        try {
+            getNodeEngine().getPartitionService().removePartitionLostListener(this.partitionLostRegistration);
+        } catch (Exception ex) {
+            if (peel(ex, HazelcastInstanceNotActiveException.class, null)
```

Comment:
I think you might get away with not having this hack by getting a reference to the partition service when constructing the `DistributedScheduledExecutorService`. But sometimes that's not possible as well if the partition service is not available at that moment.

Reply:
Getting the reference isn't the problem, doing the action is. It executes some operations internally, which trigger the exception.